tests/test_symspell.py

```diff
@@ -286,3 +286,15 @@ def test_lookup_with_canonical_term():
         suggestions = sym_spell.lookup('Test', Verbosity.CLOSEST)
         assert 1 == len(suggestions)
         assert suggestions[0].term == canonical_term
+
+
+def test_lookup_exclude_self():
+    a_list = ["badminton", "batminton", "godminton"]
+
+    sym_spell = SymSpell()
+    sym_spell.load_dictionary(ListIterator(a_list))
+    suggestions = sym_spell.lookup("badminton", Verbosity.ALL, exclude_self=True)
+
+    assert not next((sugg for sugg in suggestions if sugg.term == "badminton"), None)
+    assert next((sugg for sugg in suggestions if sugg.term == "batminton"))
+    assert next((sugg for sugg in suggestions if sugg.term == "godminton"))
```

Comment:
Unclear to me what this does. Is it the same as
```
assert len([sugg for sugg in suggestions if sugg.term == "godminton"]) > 0
```

Reply:
next will automatically take the first item in an iterable that match the condition. The optional second argument define a default value if the iterations ended without triggering the condition.

In this case I wanted to test if the exclude_self attribute I created works, this attribute make sure that if you ask for suggestion and the term you are asking for is in the dictionary, that it won't be returned. This is because I'm using SymSpell in a way that was not originally intended and I absolutely don't care about the suggestion with a distance of 0, because I already know it's in there and I don't want it (search within the domain)

By searching for badminton, I want to make sure it's not in the returned list, so a next on sugg.term == "badminton" should return None, which will fail the assert condition.

In the two other cases it's just gravy, I want to make sure both the two other terms has been suggested because SymSpell has a default distance lookup of 2, and both the two other terms are within the distance and should be suggested by SymSpell. So next() will return a string, which will by default evaluate to True with the assert.

Reply:
I could have used `any` instead of `next` I think for the same logic, might be clearer. Let me know if you think it's worth it.